### DISCUSSION TITLE
Edit functionality when there is a relationship with m2m table

### DISCUSSION THREAD

**Discussion by ovidijusku** [aminalaee/sqladmin#541] [Q&A] [answered]
Good afternoon,

I've faced strange problem which was caused by the number of records in the M2M table (to be specific, when `user_answer` record count was in tens of thousands everything worked perfectly fine, but when it reached millions - some problems occured). When the admin panel is opened up inside `UserView` (will be presented below) it lists everything properly, search works as expected. However, when edit button is clicked on any of the existing users, it stucks on the query `select user_answer.user_id, user_answer.answer_id, user_answer.metadata from user_answer` (that is really heavy one, without any filtering on specific user entity). I want to understand if that is something that could be fixed with some additional configurations / overrides? Or maybe there is something that I have missed in terms of how `ModelView` / admin panel initialization should be handled?

Using: `sqladmin == 0.13.0`

Current database schema like this:
1) `user` table with columns (id, name)
2) `answer` table with columns (id, text)
3) `user_answer` table with columns (user_id, answer_id, metadata)

user_view.py
```python
from sqladmin import ModelView

from models.user import User


class UserView(ModelView, model=User):
    can_create = False
    can_edit = True
    can_delete = False
    can_view_details = True

    column_list = [
        "id",
        "name",
    ]
    column_searchable_list = [
        "id",
        "name",
    ]
```

admin.py
```python
from fastapi import FastAPI
from sqladmin import Admin

from user_view import UserView

def init_admin(app: FastAPI) -> None:
    admin = Admin(
        app=app,
        engine=engine,
        authentication_backend=...,
        title="App admin",
    )
    admin.add_view(UserView)
```

**Comment by aminalaee** [comment, marked as answer]
So if I understand correctly it's the Edit page which is building the HTML select field and by default loads all fields. Another option is to use ajax for form options: https://aminalaee.dev/sqladmin/configurations/#form-options

**Reply by ovidijusku** [reply]
Thanks, just by excluding the relationship defined in the model from the `UserView` problem was fixed.

If anyone will encounter similar problems, that should help:
```python
class UserView(ModelView, model=User):
    can_create = False
    can_edit = True
    can_delete = False
    can_view_details = True

    column_list = [
        "id",
        "name",
    ]
    column_searchable_list = [
        "id",
        "name",
    ]
    form_excluded_columns = ["user_answers"]
    
```
